src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1479,16 +1531,61 @@ HasUnsupportedReferenceTableJoin(PlannerRestrictionContext *plannerRestrictionCo
 
 
 /*
- * RelationInfoHasReferenceTable check whether the relationInfo has reference table.
+ * RelationInfoContainsReferenceTable check whether the relationInfo
+ * has reference table.
+ *
  * Since relation ids of relationInfo indexes to the range table entry list of
  * planner info, planner info is also passed.
+ *
+ * Join relation infos are ignored since we're already going to process each of them
+ * seperately.
  */
 static bool
-RelationInfoHasReferenceTable(PlannerInfo *plannerInfo, RelOptInfo *relationInfo)
+RelationInfoContainsReferenceTable(PlannerInfo *plannerInfo, RelOptInfo *relationInfo)
```

Comment:
I don't see a big difference between `Has` and `Contains` here. Maybe we could be more descriptive about the problem we're trying to address, along the lines of `RelationInfoResultIsSharded`.

Reply:
After all the cleanup, I still would like to stick to `RelationInfoContainsReferenceTable()` since it makes more sense now?